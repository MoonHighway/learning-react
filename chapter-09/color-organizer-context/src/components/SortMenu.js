import '../../stylesheets/Menu.scss'
import PropTypes from 'prop-types'
import { sortColors } from '../actions'

const options = {
    date: "SORTED_BY_DATE",
    title: "SORTED_BY_TITLE",
    rating: "SORTED_BY_RATING"
}

const SortMenu = (props, { store }) =>
    <nav className="menu">
        <h1>Sort Colors</h1>
        {Object.keys(options).map((item, i) =>
            <a key={i}
               href="#"
               className={(store.getState().sort === options[item]) ? "selected" : null}
               onClick={e => {
                   e.preventDefault()
                   store.dispatch( sortColors(options[item]) )
               }}>{item}</a>
        )}
    </nav>

SortMenu.contextTypes = {
    store: PropTypes.object
}

export default SortMenu
