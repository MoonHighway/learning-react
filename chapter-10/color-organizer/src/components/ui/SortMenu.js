import PropTypes from 'prop-types'
import '../../stylesheets/Menu.scss'

const options = {
    date: "SORTED_BY_DATE",
    title: "SORTED_BY_TITLE",
    rating: "SORTED_BY_RATING"
}

const SortMenu = ({ sort="SORTED_BY_DATE", onSelect=f=>f}) =>
    <nav className="menu">
        <h1>Sort Colors</h1>
        {Object.keys(options).map((item, i) =>
            <a key={i}
                href="#"
                className={(sort === options[item]) ? "selected" : null}
                onClick={e => {
                    e.preventDefault()
                    onSelect(options[item])
                }}>{item}</a>
        )}
    </nav>

SortMenu.propTypes = {
    sort: PropTypes.string,
    onSelect: PropTypes.func
}

export default SortMenu
