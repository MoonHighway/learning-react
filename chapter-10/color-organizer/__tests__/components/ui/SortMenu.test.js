import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import SortMenu from '../../../src/components/ui/SortMenu'

const { shallow, mount } = Enzyme

describe("<Color /> UI Component", () => {

    let wrapper

    it("Renders Correctly", () =>
        compose(expect,toJSON,shallow)(
            <SortMenu sort="SORTED_BY_TITLE" />
        ).toMatchSnapshot()
    )

    it("click does not cause error", () => {
        mount(<SortMenu sort="SORTED_BY_TITLE" />)
            .children()
            .childAt(2)
            .simulate('click')
    })

    it("onSelect returns correct option", () => {
        const _select = jest.fn()
        const wrapper = mount(<SortMenu onSelect={_select} />)
        wrapper.children().childAt(3).simulate('click')
        expect(_select.mock.calls[0][0]).toBe("SORTED_BY_RATING")
    })

})
