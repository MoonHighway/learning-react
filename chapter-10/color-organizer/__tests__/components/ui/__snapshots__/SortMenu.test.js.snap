// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Renders Correctly 1`] = `
<nav
  className="menu"
>
  <h1>
    Sort Colors
  </h1>
  <a
    className={null}
    href="#"
    key="0"
    onClick={[Function]}
  >
    date
  </a>
  <a
    className="selected"
    href="#"
    key="1"
    onClick={[Function]}
  >
    title
  </a>
  <a
    className={null}
    href="#"
    key="2"
    onClick={[Function]}
  >
    rating
  </a>
</nav>
`;
